TokenMenu.js

```diff
@@ -2038,6 +2253,558 @@ function build_token_auras_inputs(tokens) {
+			});
+		}
+		imageSizeInputRange.val(imageSizeInput.val());
+	});
+	imageSizeInput.on('focusout', function(event) {
+		var imageSize;
+		if(event.target.value <= 6 && event.target.value >= 0.2) { 
+			imageSize = event.target.value;
+		}
+		else if(event.target.value > 6){
+			imageSize = 6;
+			imageSizeInput.val(imageSize);	
+			imageSizeInputRange.val(imageSize);
+		}
+		else if(event.target.value < 0.2){
+			imageSize = 0.2;
+			imageSizeInput.val(imageSize);	
+			imageSizeInputRange.val(imageSize);
+		}	
+		tokens.forEach(token => {
+			token.options.imageSize = imageSize;
+			token.place_sync_persist();
+		});
+
+		imageSizeInputRange.val(imageSizeInput.val());
+	});
+	imageSizeInput.on(' input change', function(){
+   	 	imageSizeInputRange.val(imageSizeInput.val());
+	});
+	imageSizeInputRange.on(' input change', function(){
+   	 	imageSizeInput.val(imageSizeInputRange.val());
+	});
+	imageSizeInputRange.on('mouseup', function(){
+   	 	let imageSize = imageSizeInputRange.val();
+		tokens.forEach(token => {
+			token.options.imageSize = imageSize;
+			token.place_sync_persist();
+		});
+	});
+	let imageSizeWrapper = $(`
+		<div class="token-image-modal-url-label-wrapper image-size-wrapper">
+			<div class="token-image-modal-footer-title image-size-title">Token Image Scale</div>
+		</div>
+	`);
+	imageSizeWrapper.append(imageSizeInput); // Beside Label
+	imageSizeWrapper.append(imageSizeInputRange); // input below label
+	body.append(imageSizeWrapper);
+
+	//border color selections
+	let borderColorInput = $(`<input class="border-color-input" type="color" value="#ddd"/>`);
+	let tokenBorderColors = tokens.map(t => t.options.color);
+	if(tokenBorderColors.length === 1) {
+		borderColorInput.val(tokenBorderColors[0] || "#dddddd");	
+	}
+	let borderColorWrapper = $(`
+		<div class="token-image-modal-url-label-wrapper border-color-wrapper">
+			<div class="token-image-modal-footer-title border-color-title">Border Color</div>
+		</div>
+	`);
+	borderColorWrapper.append(borderColorInput); 
+	body.append(borderColorWrapper);
+	let colorPicker = $(borderColorInput);
+	colorPicker.spectrum({
+		type: "color",
+		showInput: true,
+		showInitial: true,
+		containerClassName: 'prevent-sidebar-modal-close',
+		clickoutFiresChange: true,
+		color: tokens[0].options.color,
+		appendTo: "parent"
+	});
+	const borderColorPickerChange = function(event, tinycolor) {
+		let borderColor = `rgba(${tinycolor._r}, ${tinycolor._g}, ${tinycolor._b}, ${tinycolor._a})`;
+		if (event.type === 'change') {
+			tokens.forEach(token => {
+				token.options.color = borderColor;
+				$("#combat_area tr[data-target='" + token.options.id + "'] img[class*='Avatar']").css("border-color", borderColor);
+				token.place_sync_persist();
+			});
+		}
+		else {
+			tokens.forEach(token => {
+				let selector = "div[data-id='" + token.options.id + "']";
+				let html = $("#tokens").find(selector);
+				let options = Object.assign({}, token.options);

```

Comment:
html and options supposed to be assigned by not used?

Reply:
This is probably part of clean up to do here and there that was missed if they aren't being used. There's definitely clean up with removing the old context menu code.

Reply:
I'll clean this up later today - probably copy paste mistake

Reply:
This should be fixed now as should the fading option toggles